pkg/stats/collector.go

```diff
@@ -245,6 +244,9 @@ func (s *BufferedCollector) isCtxCancelled() bool {
 }
 
 func (s *BufferedCollector) Run(ctx context.Context) {
+	if atomic.LoadInt32(&s.runCalled) != 0 {
```

Comment:
Why would run be called more than once?

Reply:
Also consider using `Sync.Once`.

Reply:
Will do. I also think it is bit over-eng - but will address it in a different issue.

Reply:
sync.Once can help making sure `Run` will perform per incese. Currently the counter also used a marker for 'Close' to indicate we do not close/wait for resources we allocated on Run.